demos/qml/tutorial_learningshallow.ipynb

```diff
@@ -0,0 +1,1003 @@
+      "File \u001b[0;32m~/Workspace/catalyst-clone/frontend/catalyst/compiled_functions.py:161\u001b[0m, in \u001b[0;36mCompiledFunction._exec\u001b[0;34m(shared_object, has_return, numpy_dict, *args)\u001b[0m\n\u001b[1;32m    159\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m shared_object \u001b[38;5;28;01mas\u001b[39;00m lib:\n\u001b[1;32m    160\u001b[0m     result_desc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mtype\u001b[39m(args[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mcontents) \u001b[38;5;28;01mif\u001b[39;00m has_return \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m--> 161\u001b[0m     retval \u001b[38;5;241m=\u001b[39m \u001b[43mwrapper\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwrap\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlib\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfunction\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mresult_desc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlib\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmem_transfer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnumpy_dict\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    163\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m retval\n",
+      "\u001b[0;31mRuntimeError\u001b[0m: [/home/tzung-han.juang/Workspace/catalyst-clone/runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp][Line:238][Function:State] Error in Catalyst Runtime: Invalid size for the pre-allocated state vector"
+     ]
+    }
+   ],
+   "source": [
+    "def V_0():\n",
+    "    qml.CNOT((0, 1))\n",
+    "    qml.CNOT((1, 2))\n",
+    "    qml.Hadamard(0)\n",
+    "\n",
+    "dev = qml.device(\"lightning.qubit\", wires=n*2)\n",
+    "\n",
+    "# Not supported\n",
+    "@qjit\n",
+    "@qml.qnode(dev)\n",
+    "def local_inversion():\n",
+    "    U_test()   # some shallow unitary circuit\n",
+    "    V_0()      # supposed to disentangle qubit 0\n",
+    "    return qml.density_matrix(wires=[0])\n",
```

Comment:
Hi @josh146, according to #826, density matrix measurement does not work with qjit. And we have several examples in this demo using it. Should we still keep these tests?

Reply:
@tzunghanjuang and I spoke offline, and I suggested doing
```python
   # return qml.density_matrix(wires=wires)
   return qml.state()

state = circuit(...)
dm = qml.math.dm_from_statevector(state)
dm = qml.math.reduce_dm(density_matrix, indices=wires)
```